hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -85,6 +85,9 @@ object InferPType {
 
         val allRequired = start.pType2.required && stop.pType2.required && step.pType2.required
         PStream(start.pType2.setRequired(true), allRequired)
+      case ArrayZeros(length) =>
+        infer(length)
+        PArray(PInt32())
```

Comment:
I think the PInt32() can be required (because if the PArray is constructed it will have only non-null values), and that the PArray's requiredeness can be that of the length (if length provided the array will be constructed).

Lastly, I think we should use the concrete constructor where possible (PCanonicalArray in this case); the PArray constructors will go away.

Reply:
This is the only thing that needs a fix.